libbindgen/src/regex_set.rs

```diff
@@ -19,48 +19,45 @@ impl RegexSet {
     }
 
     /// Extend this set with every regex in the iterator.
-    pub fn extend<I>(&mut self, iter: I)
-        where I: IntoIterator<Item = String>,
+    pub fn extend<I, S>(&mut self, iter: I)
+        where I: IntoIterator<Item = S>,
+              S: Into<String>
     {
         for s in iter.into_iter() {
-            self.insert(&s)
+            self.insert(s)
         }
     }
 
     /// Insert a new regex into this set.
-    pub fn insert<S>(&mut self, string: &S)
-        where S: Borrow<str>,
+    pub fn insert<S>(&mut self, string: S)
+        where S: Into<String>
     {
-        let s = string.borrow();
-        match Regex::new(&format!("^{}$", s)) {
-            Ok(r) => {
-                self.items.push(r);
-            }
-            Err(err) => {
-                warn!("Invalid pattern provided: {}, {:?}", s, err);
-            }
+        let s = string.into();
+        self.items.push(s);
+        self.set = None;
+    }
+
+    pub fn build(&mut self) {
+        self.set = match RxSet::new(&self.items) {
```

Comment:
Also, I believe we need to make the patterns exact, see how before this patch we used `Borrow<str>` with `format!("^{}$")`.

This should make travis happy again I think.